### PR TITLE
add jest + react-testing library with example and docs guide

### PR DESCRIPTION
**Pull request by balzss**
INSTUI-3810

- add jest + react testing library to the project
- add rules, config and script to run them
- run with CI
- convert `ui-avatar` tests as example and skip old ones
- add guide to docs about converting tests

test plan
- run new tests with `yarn test:new` -> all 26 should pass
- run old `ui-avatar` tests with `npx ui-scripts test --path packages/ui-avatar` -> all 26 should be skipped (not including the theme related ones)
- read the guide about tests [here](https://1246--preview-instui.netlify.app/#converting-old-tests)

### CODE REVIEW

File: packages/ui-avatar/package.json [hunk 1]
```diff
@@ -34,11 +34,14 @@
     "prop-types": "^15.8.1"
   },
   "devDependencies": {
+    "@instructure/ui-axe-check": "workspace:^",
```
Comment (matyasf) [inline packages/ui-avatar/package.json:37]: We should use here exact dependency numbers like with other internal references

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/** @type {import('ts-jest').JestConfigWithTsJest} */
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'jsdom',
+  testMatch: ['**/__new-tests__/**/*.test.tsx']
+}
```
Comment (matyasf) [inline jest.config.js:29]: We should calculate test coverage too (and later fail the build if its below 80%)

**Review by matyasf — CHANGES_REQUESTED** [review]
looking good, just some smaller issues

File: jest.config.js (hunk 2 above)
Comment (balzss) [reply jest.config.js:37]: I've added the coverage threshold to the config. Once we are ready, we can add the `--coverage` flag to the script and it should work.

**Review by matyasf — APPROVED** [review]

**Review by HerrTopi — APPROVED** [review]

**Pull request merged**